File: Python/long_pal_substr.py
s = input()
n = len(s)
res = ""
resl = 0
for i in range(n):
    l,r = i,i
    while(l>= 0 and r<n and s[l] == s[r]):
        if(r-l+1 > resl):
            res = s[l:r+1]
            resl = r-l+1
        l -= 1
        r += 1
    l,r=i,i+1
    while(l>=0 and r<n and s[l]==s[r]):
        if(r-l+1 > resl):
            res = s[l:r+1]
            resl = r-l+1
        l -= 1
        r += 1
print(res)

File: C++/long_pal_substr.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    string s;
    getline(cin,s);
    int n = s.size();
    string res = "";
    int resL = 0;
    for(int i=0; i<n; i++){
        int l=i,r=i;
        while(l>=0 && r<n && s[l] == s[r]){
            if(resL < r-l+1){
                resL = r-l+1;
                res = s.substr(l,resL);
            }
            l--;
            r++;
        }
        l=i, r=i+1;
        while(l>=0 && r<n && s[l] == s[r]){
            if(resL < r-l+1){
                resL = r-l+1;
                res = s.substr(l,resL);
            }
            l--;
            r++;
        }
    }
    cout << res;
}